### DISCUSSION TITLE
The 'WinRT.Interop.WindowNative.GetWindowHandle' call is causing an exception

### DISCUSSION THREAD

**Discussion by frankayars** [unoplatform/uno#10530] [Q&A]
On the Uno Widnows platform project, the following call always results in an exception:

IntPtr hWnd = WinRT.Interop.WindowNative.GetWindowHandle(this);

The exception message is 'The specified cast is not valid'. The issue is not limited to the parent control. Any UIElement based object passed into that call will cause the exception. Our current project requires the underlying hWnd handle to integrate with 3rd party 3D graphics APIs and any information to resolve this exception would be appreciated.

**Comment by jeromelaban** [comment]
Thanks for the report. This is a question to be [asked in the WinUI repository](https://github.com/microsoft/microsoft-ui-xaml/discussions), as Uno.UI is not present in this target.

**Comment by frankayars** [comment]
I will post to that repository discussion. Thank you.
